### ISSUE TITLE
[SPIKE] Make VAMC, Benefit detail pages, and Resources & Support On this page ToC dependent on two h2s, instead of opt-in

### ISSUE THREAD

**Issue by kevwalsh** [department-of-veterans-affairs/va.gov-cms#9602]
## Questions
- How to find the list of pages using the ToC?
- Is this a Drupal ticket, or FE? Or do I need one of each?
  - [Fran] I've turned this ticket into a Spike, and have updated the AC and description. The outcome may be that we'll have two tickets, one Drupal and one FE. 

## User Story or Problem Statement

We believe that this is affecting Benefit Detail pages, Resources & Support detail pages, and VAMC detail pages. This ticket is a spike to discover which pages actually use the 'Include Table of Contents' feature, which can then results in the page displaying an "On this page' heading with no H2 links under it.

The outcome of this spike (as indicated in the AC), is to discover all the pages which use this feature, and then create a list of the pages. 

Note that if the outcome is only VAMC pages utilize and wind up in this state, we may then share the results with the FAC team so tickets can be created in their project.

## Information

On Benefit Detail pages, and VAMC Detail pages, editors can currently choose whether or not to include a table of contents using this checkbox.

<img width="939" alt="Create_VAMC_Detail_Page___VA_gov_CMS" src="https://user-images.githubusercontent.com/643678/176189289-476c34ab-3dfd-4eb4-b2f4-c55012ac46fd.png">

### Example URLs:

- https://www.va.gov/pittsburgh-health-care/health-services/surgery-service/
- https://www.va.gov/boise-health-care/work-with-us-/volunteer-or-donate/

_Note that I was unable to find examples of Benefit Detail pages nor Resources & Support pages._

When this component is used incorrectly by a VAMC editor, the design system component shows an "On this page" header, with no links. Note that there are some cases where there aren't any h2s on the page, because the editor has added only h3s.

<img width="590" height="326" alt="Image" src="https://github.com/user-attachments/assets/223c2da5-12be-4cc5-a1b5-9f7163807f17" />

This is a [known issue that will not be resolved at the design system component level](https://github.com/department-of-veterans-affairs/va.gov-team/issues/28249#) https://github.com/department-of-veterans-affairs/va.gov-team/issues/28249#. 

There are a half dozen content types / page templates in the VAMC product for which the ToC is not optional. 

This task is to make the Benefit Detail page and VAMC detail page ToC dependent on the existence of two h2s.

## Acceptance Criteria
- [ ] Determine which pages use the 'Include Table of Contents' Drupal feature
- [ ] Create a list of all the pages
- [ ] Ping @FranECross so that she can create any follow-up tickets for the work needed
- [ ] Close ticket

**Comment by davidmpickett** [comment]
This issue was surfaced again: https://dsva.slack.com/archives/C0FQSS30V/p1673645534388259

**Comment by laflannery** [comment]
I would want this done other places this is used as well. 2 of those places are Benefits Detail pages and Resources and Support pages, both of which are PW products. Because this crosses both teams, is this a CMS issue OR is this actually something that has to be done per template/content type so I should make a ticket for PW for their products?

**Comment by mmiddaugh** [comment]
@laflannery can you give this a defect level to assist with prioritization, please?

**Comment by r-bartlett-gsa** [comment]
@FranECross Can you please confirm if this is still an issue and would your team take it. Please take a look at Laura's comment above.

If it's not an issue, please close it.

**Comment by FranECross** [comment]
This needs more research to see where all this is happening. We may need drupal and FE tickets to follow up. Will turn this into a spike. @FranECross Alter description to have this be a Spike; change ac to have me create tickets after the spike is done.
